docs/rendering_system.rst

```diff
@@ -0,0 +1,241 @@
+..
+   Author: Adam Turner
+
+   We can't use :pep:`X` references in this document, as they are relative from
+   the content root.
+   We use :doc:`PEP X <../pep-xxxx>` instead.
+
+
+An Overview of the PEP Rendering System
+=======================================
+
+This document provides an overview of the PEP rendering system, as a companion
+to :doc:`PEP 676 <../pep-0676>`.
+
+
+1. Configuration
+----------------
+
+Configuration is held in three files:
+
+- ``conf.py`` contains the majority of the configuration
+- ``contents.rst`` contains the table of contents directives for Sphinx
+- ``pep_sphinx_extensions/pep_theme/theme.conf`` sets the Pygments themes
+
+The configuration:
+
+- registers the custom Sphinx extension
+- sets both ``.txt`` and ``.rst`` suffixes to be parsed as PEPs
+- tells Sphinx which source files to use
+- registers the PEP theme, maths renderer, and template
+- disables some default settings that are covered in the extension
+- sets the default and 'dark mode' code formatter styles
+
+
+2. Orchestration
+----------------
+
+``build.py`` acts to manage the rendering process.
+Usage is covered in :doc:`build`.
+
+
+3. Extension
+------------
+
+The Sphinx extension and theme are contained in the ``pep_sphinx_extensions``
+directory.
+The following is a brief overview of the stages of the PEP rendering process,
+and how the extension functions at each point.
+
+
+3.1 Extension setup
+~~~~~~~~~~~~~~~~~~~
+
+The extension registers several objects:
+
+- ``FileBuilder`` and ``DirectoryBuilder`` run the build process for file- and
+  directory- based building
+- ``PEPParser`` registers the custom document transforms and parses PEPs to
+  a Docutils document.
+- ``PEPTranslator`` converts a Docutils document into HTML
+- ``PEPRole`` handles ``:pep:`` roles in the reStructuredText source
+
+It also patches default behaviour:
```

Comment:
```suggestion
The extension also patches default behaviour:
```

At first I was uncertain what "it" referred to, since we are many lines away from where it was defined.